bin/cli.class.php

```diff
@@ -430,6 +430,14 @@ public function run() {
         continue;
       }
       $this->row++;
+      if ($this->row % 1000 == 0) {
+        // Reset PEAR_DB_DATAOBJECT cache to prevent memory leak
+        $GLOBALS['_DB_DATAOBJECT']['RESULTS'] = array();
```

Comment:
@mickadoo No side effects in this scenario, no.  However, I can't guarantee there would be no side effects Civi-wide, so I don't think disabling the cache is a worthwhile endeavor.  

This object is the one that `CRM_Core_DAO` is extending, so it's very low-level.  While there DOES appear to be an inherited class called [DB_DataObject_Overload](https://pear.php.net/package/DB_DataObject/docs/latest/apidoc/DB_DataObject/DB_DataObject_Overload.html) that we could extend instead, I think we'd find it rather difficult to only disable the cache when it was relevant.

@totten @eileenmcnaughton If you're more familiar with the `DB_DataObject->_resultFields` property than I am, please feel free to comment.